src/frontend/screens/Settings/MapSettings/BackgroundMapInfo.tsx

```diff
@@ -0,0 +1,330 @@
+    id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail19",
+    defaultMessage: "Local highway and crossing details",
+  },
+  lvlOfDetail20: {
+    id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail20",
+    defaultMessage: "A mid-sized building",
+  },
+});
+
+// To Do: Get level of detail from programs team
+export const BackgroundMapInfo = ({
+  route,
+  navigation,
+}: NativeRootNavigationProps<"BackgroundMapInfo">) => {
+  const { formatMessage: t } = useIntl();
+  const { bytesStored, id, styleUrl, name } = route.params;
+  const [zoom, setZoom] = React.useState<"loading" | number | null>("loading");
+
+  const sheetRef = React.useRef<BottomSheetMethods>(null);
+
+  const levelOfDetail = React.useMemo(() => {
+    if (typeof zoom !== "number") return undefined;
+
+    switch (true) {
+      case zoom >= 0 && zoom < 2:
+        return m.lvlOfDetail0to1;
+      case zoom === 2:
+        return m.lvlOfDetail2;
+      case zoom >= 3 && zoom < 5:
+        return m.lvlOfDetail3to4;
+      case zoom === 5:
+        return m.lvlOfDetail5;
+      case zoom === 6:
+        return m.lvlOfDetail6;
+      case zoom >= 7 && zoom < 9:
+        return m.lvlOfDetail7to8;
+      case zoom === 9:
+        return m.lvlOfDetail9;
+      case zoom === 10:
+        return m.lvlOfDetail10;
+      case zoom === 11:
+        return m.lvlOfDetail11;
+      case zoom === 12:
+        return m.lvlOfDetail12;
+      case zoom >= 13 && zoom < 15:
+        return m.lvlOfDetail13to14;
+      case zoom === 15:
+        return m.lvlOfDetail15;
+      case zoom === 16:
+        return m.lvlOfDetail16;
+      case zoom === 17:
+        return m.lvlOfDetail17;
+      case zoom === 18:
+        return m.lvlOfDetail18;
+      case zoom === 19:
+        return m.lvlOfDetail19;
+      case zoom === 20:
+        return m.lvlOfDetail20;
+      default:
+        return undefined;
+    }
+  }, [zoom]);
+
+  const { setStyleId } = useMapStyle();
+
+  function setStyleAndNavigateHome() {
+    setStyleId(id);
+    navigation.navigate("Home", { screen: "Map" });
+  }
+
+  React.useEffect(() => {
+    api.maps
+      .getStyle(id)
+      .then(style => {
+        const tileSetIdArray = Object.values(style.sources).map(source => {
+          if ("url" in source && source.url) {
+            const url = source.url;
+            const searchWord = "tilesets/";
+            const startPosition = url.search(searchWord);
+            if (startPosition === -1) return;
+            const endPosition = url.length;
+            return url.substring(
+              startPosition + searchWord.length,
+              endPosition
+            );
+          }
+        });
+
+        // Promise.allSettled does not work in React Native
+        // See https://github.com/facebook/react-native/issues/30236
+        return allSettled(
+          tileSetIdArray.map(id => {
+            if (id) {
+              return api.maps.getTileset(id);
+            }
+          })
```

Comment:
think you can simplify this to not check if the id exists or not. if it doesn't, you're most likely dealing with an empty array `[]` as opposed to an array of undefined e.g. `[undefined, undefined]`. Mapping over an empty array will produce an empty array anyways

Reply:
This is actually related to the (allSettled)[https://github.com/digidem/mapeo-mobile/pull/1029#discussion_r1012203260] comment below. When creating`tileSetIdArray`, typescript is recognizing it as `(string|undefined)[]`.  Technically it should be a `string[]` || `[]`. Perhap I am missing something, but I believe I am creating `tileSetIdArray` to only be `string[]` || `[]` and typescript is equating `[]` to `undefined[]`.

Reply:
ah yeah. instead of mapping over the object values i.e. `Object.values(style.sources).map...`, i would use a for loop i.e.:

```ts
const tilesetIds: string[] = []

for (const source of Object.values(style.sources) {
  // do the url logic stuff and get the tileset id
  
  // push to the array
  tilesetIds.push(id)
}

return allSettled(tilesetIds.map(id => api.getTileset(id))
```

this guarantees that the ids in `tilesetIds` are never undefined (you're only pushing string values to it)